cores/Main.cpp

```diff
@@ -41,8 +41,9 @@ int main(void)
  */
 wiring_time_init();
 wiring_analog_init();
-// Initialize the reset pin for the XMC1100 Boot Kit series
-#ifdef XMC1100_Boot_Kit
+// Initialize the reset pin for the XMC1100 Boot Kit series and XMC1400 Kit for Arduino as they are based on Arduino form-factor
+// Hence, a dedicated reset pin is required. 
+#if defined(XMC1100_Boot_Kit) || defined(XMC1400_Arduino_Kit)
```

Comment:
Change to board agnostic method here using define in pins_arduino.h (also on XMC1100 Boot Kit)

#define HAS_GPIO_RESET

so 
#ifdef HAS_GPIO_RESET

Reply:
Currently, only two boards XMC 1100 Boot Kit and XMC 1400 Kit for Arduino have dedicated reset pins. Such a boad agnostic approach may be added when more boards with dedicated resets are added to the core.

Reply:
NO **NOW** is the **TIME** to do it BEFORE another board comes along

Obviously you think of this as a student exercise and everybody else is going to have to TIDY UP your **mess** later

Not resolved someone trying to do bare minimum to pass an exam